### PR TITLE
✨Document running E2E tests locally

### PR DESCRIPTION
**Pull request by mdbooth**
**What this PR does / why we need it**:

Documents how to run E2E tests locally, including the enhancements added in #981.

### CODE REVIEW

File: docs/book/src/development/development.md [hunk 1]
```diff
@@ -42,3 +42,103 @@ After generating `infrastructure-components.yaml`, replace the `us.gcr.io/k8s-ar
...
+
+You *MUST* configure systemd and iptables as described in https://kind.sigs.k8s.io/docs/user/rootless/. There is no need to configure cgroups v2 as Fedora uses this by default.
+
+You must install the `podman-docker` package to emulate the docker cli tool. However, this is not sufficient on its own as described below.
+
+### Running podman system service to emulate docker daemon
+
+While kind itself supports podman, the cluster-api test framework does not. This framework is used by the CAPO tests to push test images into the kind cluster. Unfortunately the cluster-api test framework explicitly connects to a running docker daemon, so cli emulation is not sufficient for compatibility. This issue is tracked in https://github.com/kubernetes-sigs/cluster-api/issues/5146, and the following workaround can be ignored when this is resolved.
+
+podman includes a 'system service' which emulates docker. For the tests to work, this service must be running and listening on a unix socker at `/var/run/docker.sock`. You can achieve this with:
```
Comment (stmcginnis) [inline docs/book/src/development/development.md:138]: s/socker/socket/

**Review by stmcginnis — COMMENTED** [review]
Great info! Just noticed one small typo.

File: docs/book/src/development/development.md (hunk 1 above)
Comment (mdbooth) [reply docs/book/src/development/development.md]: D'oh! Fixed.

File: docs/book/src/development/development.md [hunk 2]
```diff
@@ -42,3 +42,103 @@ After generating `infrastructure-components.yaml`, replace the `us.gcr.io/k8s-ar
...
+### Support for clouds using SSL
+
+If your cloud requires a cacert you must also pass this to make via `OPENSTACK_CLOUD_CACERT_B64`, i.e.:
+
+```bash
+make test-e2e OPENSTACK_CLOUD_YAML_FILE=/path/to/clouds.yaml OPENSTACK_CLOUD=my_cloud \
+              OPENSTACK_CLOUD_CACERT_B64=$(base64 -w /path/to/mycloud-ca.crt)
+```
+
+CAPO deployed in the local kind cluster will automatically pick up a `cacert` defined in your `clouds.yaml` so you will see servers created in OpenStack without specifying `OPENSTACK_CLOUD_CACERT_B64`. However, the cacert won't be deployed to those servers, so kubelet will fail to start.
```
Comment (jichenjc) [inline docs/book/src/development/development.md:73]: not sure I fully understand this ,the cacert is mostly used for openstack interaction defined in clouds.yaml
so why it's related to kubelet?

File: docs/book/src/development/development.md [hunk 3]
```diff
@@ -42,3 +42,103 @@ After generating `infrastructure-components.yaml`, replace the `us.gcr.io/k8s-ar
...
+To do this you need to configure the host appropriately and pass `PODMAN=1` to make, i.e.:
+
+```bash
+make test-e2e OPENSTACK_CLOUD_YAML_FILE=/path/to/clouds.yaml OPENSTACK_CLOUD=my_cloud \
+              PODMAN=1
+```
+
+### Host configuration
+
+Firstly, you must be using kernel >=5.11. Fedora 34 is the oldest distribution with this kernel.
```
Comment (jichenjc) [inline docs/book/src/development/development.md:128]: oldest ? latest?

File: docs/book/src/development/development.md [hunk 4]
```diff
@@ -42,3 +42,103 @@ After generating `infrastructure-components.yaml`, replace the `us.gcr.io/k8s-ar
...
+| ID                                   | Name     | Subnets                              |
++--------------------------------------+----------+--------------------------------------+
+| 27635f93-583d-454e-9c6d-3d305e7f8a22 | external | be64cd07-f8b7-4705-8446-26b19eab3914 |
+| cf2e83dc-545d-490f-9f9c-4e90927546f2 | hostonly | ec95befe-72f4-4af6-a263-2aec081f47d3 |
++--------------------------------------+----------+--------------------------------------+
+```
+
+### OpenStack prerequisites
+
+The file `test/e2e/data/e2e_conf.yaml` and the test templates under `test/e2e/data/infrastructure-openstack` reference several OpenStack resources which must exist before running the test:
```
Comment (jichenjc) [inline docs/book/src/development/development.md:100]: nit: I think we can provide link of the conf and template files

File: docs/book/src/development/development.md (hunk 3 above)
Comment (mdbooth) [reply docs/book/src/development/development.md:128]: oldest is correct here. It's also the latest! However, this statement will still be correct when it is no longer the oldest.

File: docs/book/src/development/development.md (hunk 2 above)
Comment (mdbooth) [reply docs/book/src/development/development.md:73]: We also need to pass the cacert into the cluster so the cloud provider [1] can talk to OpenStack. If the cloud provider fails to come up then the Node will not become ready and the tests fail.

[1] kubelet in-tree provider, or CCM for external cloud provider

Comment (jichenjc) [reply docs/book/src/development/development.md:73]: ok, OCCM need this cacert, thanks for the reminder

**Pull request merged**